core/src/main/java/org/pentaho/di/core/row/ValueDataUtil.java

```diff
@@ -398,6 +415,56 @@ public static String createChecksum( ValueMetaInterface metaA, Object dataA, Str
     return md5Hash;
   }
 
+  public static String createChecksum( ValueMetaInterface metaA, Object dataA, String type, boolean failIfNoFile )
+          throws KettleFileException {
+    if ( dataA == null ) {
+      if ( failIfNoFile ) {
+        throw new KettleFileException();
+      }
+      return null;
+    }
+
+    String md5Hash = null;
+    FileObject file = null;
+    try {
+      file = KettleVFS.getFileObject( dataA.toString() );
+      throwsErrorOnFileNonExistence( file );
+      InputStream in = KettleVFS.getInputStream( file );
+      int bytes = in.available();
+      byte[] buffer = new byte[bytes];
+      in.read( buffer );
+
+      StringBuffer md5HashBuff = new StringBuffer( 32 );
+      byte[] b = MessageDigest.getInstance( type ).digest( buffer );
+      int len = b.length;
+      for ( int x = 0; x < len; x++ ) {
+        md5HashBuff.append( String.format( "%02x", b[x] ) );
+      }
+
+      md5Hash = md5HashBuff.toString();
+
+    } catch ( KettleFileException e ) {
+      if ( failIfNoFile ) {
+        throw e;
+      }
+    } catch ( Exception e ) {
+      // ignore - should likely log the exception
+    } finally {
+      if ( file != null ) {
+        try {
+          file.close();
+        } catch ( Exception e ) {
+          // Ignore
+        }
+      }
+    }
+    return md5Hash;
+  }
+
+  /**
+   * @deprecated Use {@link ValueDataUtil#ChecksumCRC32(ValueMetaInterface, Object, boolean)} instead
+   */
+  @Deprecated
```

Comment:
shouldn't keep repeated code, the old api should call a version of the new implementations